### DISCUSSION TITLE
✨ daggerverse publishing - use package manager version info

### DISCUSSION THREAD

**Discussion by rawkode** [dagger/dagger#12166] [Ideas]
### What are you trying to do?

When we write a Dagger module, we need a `package.json`, `go.mod`, etc.

These files contain version information that can be consumed by the Daggerverse publishing process.

Currently, we're expected to use Git tags, which when used in a repository housing many modules is far from ideal; mostly because Git and GitHub just don't expect developers to work in this paradigm.

I'd like to see the package manager file used and mapped to a SHA rather than littering a Git repository with tags.

**Comment by sagikazarmark** [comment]
> When we write a Dagger module, we need a `package.json`, `go.mod`, etc.

I don't think `go.mod` has a version in it. Or does it?

> Currently, we're expected to use Git tags, which when used in a repository housing many modules is far from ideal; mostly because Git and GitHub just don't expect developers to work in this paradigm.

Go modules seem to have adopted this pattern and while it's not perfect, I've seen it work well enough.
